legacy_packages/sdk/src/evm/core/classes/contract-publisher.ts

```diff
@@ -406,11 +406,18 @@ export class ContractPublisher extends RPCConnectionHandler {
         "DynamicContract",
         features,
       );
+      const isModular = isFeatureEnabled(
+        compilerMetadata.abi,
+        "ModularCore",
+        features,
+      );
       extraMetadataCleaned.routerType = isPlugin
         ? "plugin"
         : isDynamic
           ? "dynamic"
-          : "none";
+          : isModular
+            ? "modular"
+            : "none";
```

Comment:
nit: starting to become unreadable

Reply:
Leaving it unchanged for now. Can clean it up when we discontinue plugin pattern.